app/src/main/java/com/example/criminalintent/CrimeRepository.kt

```diff
@@ -0,0 +1,34 @@
+package com.example.criminalintent
+
+import android.content.Context
+import androidx.lifecycle.LiveData
+import androidx.room.Room
+import com.example.criminalintent.database.CrimeDatabase
+import java.util.UUID
+
+class CrimeRepository private constructor(context: Context) {
+
+    private val database =
```

Comment:
I saw it's added here in the book too but I don't think repositories should create the database. Once it's created the same instance should be re-used and passed to other classes(if it will be needed). The repo also doesn't need access to the `context` or the entire `CrimeDatabase`, just to `crimeDao` that it uses.
I'd suggest to extract this part to `CriminalIntentApplication` before `CrimeRepository` is initialized and just pass `database.crimeDao()` to the repo